synapseclient/extensions/curator/schema_generation.py

```diff
@@ -5295,14 +5416,20 @@ def get_component_json_schema(
         Raises:
             May raise errors if the component is not found in the data model graph.
         """
-
+        metadata_model = MetadataModel(
+            inputMModelLocation=self.data_model_source,
+            inputMModelLocationType="local",
+            data_model_labels=data_model_labels,
+            logger=self.logger,
+        )
         use_display_names = data_model_labels == "display_label"
 
         json_schema = create_json_schema(
             dmge=self.dmge,
             datatype=self.component,
             logger=self.logger,
             schema_name=self.component + "_validation",
+            jsonld_path=metadata_model.inputMModelLocation,
```

Comment:
this is part of the original function in schematic: https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic/schemas/json_schema_component_generator.py#L257